### DISCUSSION TITLE
Godot 4 update

### DISCUSSION THREAD

**Discussion by Kallel181** [lennart-finke/GodoTeX#6] [Ideas]
Would be really nice to have this plugin working on Godot 4, and i think it wouldn't be many changes on it.

**Comment by lennart-finke** [comment]
Sure, I'm on it!

**Comment by lennart-finke** [comment]
Alright, so it looks like it should be possible to port everything. 

Godot 4.0 gives me quite a bit of trouble with issues not directly related to the plugin though. The ever-present `Transient parent has another exclusive child.` renders activating the addon impossible, which it looks like is getting fixed with https://github.com/godotengine/godot/pull/73365 in Godot 4.1?

So if I'm not mistaken about that, I will wait until that is sorted out.

Also, my Linux machine seemingly does not support Vulkan and executing with OpenGL3 has not been working either. Maybe that will resolve itself, too.

Until then, if somebody else is keen on porting the plugin to Godot 4, feel free to!

**Reply by BlazeKowe** [reply]
I trying to do it myself
I have done some work
but I'm not confident about it yet

**Reply by BlazeKowe** [reply]
I can get the plugin to build and load
but it isn't working correctly in the sample
I'll post some news when I do some more research

**Reply by lennart-finke** [reply]
Hi @BlazeKowe! That is very kind of you, thanks. I have tried again with 4.0.3 myself, but still without success. Please do keep us updated :^)

**Reply by martazvch** [reply]
Hi there, have you been able to acheive the Godot 4 update?

**Reply by lennart-finke** [reply]
Hello @MartinVacheron and thank you for your interest as well. It looks like Godot 4.1 already plays a bit nicer than 4.0 judging by my first look at it. I will see if I can make some progress tomorrow.

**Comment by lennart-finke** [comment]
OK folks, problem solved. I'll commit and update the Godot Add-On repository shortly. Hope you enjoy GodoTeX for Godot 4!

![Unbenannt](https://github.com/file-acomplaint/GodoTeX/assets/42694673/996c6464-fcae-49a0-a7a2-00388cd8c29c)

**Reply by martazvch** [reply]
Thank you!!

Edit: just tried it, it works juste fine in Godot_4.2_dev2 on Windows. Thanks again!
